docs/pipelines.md

```diff
@@ -516,6 +517,184 @@ In particular:
+
+You can specify a list of one or more final tasks under `finally` section. Final tasks are guaranteed to be executed
+after all `PipelineTasks` under `tasks` have completed regardless of success or error. Final tasks are very similar to
+`PipelineTasks` under `tasks` section and follow the same syntax. Each final task must have a
+[valid](https://kubernetes.io/docs/concepts/overview/working-with-objects/names/#names) `name` and a `taskRef` or
+`taskSpec`. For example:
+
+```yaml
+spec:
+  tasks:
+    - name: tests
+      taskRef:
+        Name: integration-test
+  finally:
+    - name: cleanup-test
+      taskRef:
+        Name: cleanup
+```
+
+### Specifying `Resources` in Final Tasks
+
+Similar to `tasks`, you can use [PipelineResources](#specifying-resources) as inputs and outputs for
+final tasks in the Pipeline. The only difference here is, final tasks with an input resource can have a `from` clause
+but that resource has to rely on output of a `PipelineTask` from `tasks` section. For example:
+
+```yaml
+spec:
+  tasks:
+    - name: tests
+      taskRef:
+        Name: integration-test
+      resources:
+        inputs:
+          - name: source
+            resource: tektoncd-pipeline-repo
+        outputs:
+          - name: workspace
+            resource: my-repo
+  finally:
+    - name: clear-workspace
+      taskRef:
+        Name: clear-workspace
+      resources:
+        inputs:
+          - name: workspace
+            resource: my-repo
+            from:
+              - tests
+```
+
+### Specifying `Parameters` in Final Tasks
+
+Again, similar to `tasks`, you can specify [`Parameters`](tasks.md#specifying-parameters):
+
+```yaml
+spec:
+  tasks:
+    - name: tests
+      taskRef:
+        Name: integration-test
+  finally:
+    - name: report-results
+      taskRef:
+        Name: report-results
+      params:
+        - name: url
+          value: $(params.url)
+```
+
+### Configuring `Task` execution results with `finally`
+
+Final tasks can not be configured to consume `Results` of `PipelineTask` from `tasks` section i.e. the following
+example is not supported right now but we are working on adding support for the same.
+
+```yaml
+spec:
+  tasks:
+    - name: count-comments-before
+      taskRef:
+        Name: count-comments
+    - name: add-comment
+      taskRef:
+        Name: add-comment
+    - name: count-comments-after
+      taskRef:
+        Name: count-comments
+  finally:
+    - name: check-count
+      taskRef:
+        Name: check-count
+      params:
+        - name: before-count
+          value: $(tasks.count-comments-before.results.count)
+        - name: after-count
+          value: $(tasks.count-comments-after.results.count)
+```
+
+In this example, container logs for `check-count` might have something like this: 
+
+```
+/tekton/scripts/script-0-zsg5v: 3: tasks.sum-inputs.results.sum-result: not found
+/tekton/scripts/script-0-2g2dr: 3: tasks.count-comments-before.results.count: not found
+```
```

Comment:
The header of this part make me think it's possible. I think this part should be rewritten differently, like having a `know limitation` header and list what won't work

Reply:
Also, I'm sure this has been discussed before, but what is the reason it wouldn't/couldn't support using results from previous tasks ?

Reply:
yup, we did discuss in the [API workflow group](https://docs.google.com/document/d/17PodAxG8hV351fBhSu7Y_OIPhGTVgj6OJ2lPphYYRpU/edit) 😄 

Sure no worries @vdemeester 

Its not supported right now in this PR, we discussed a few options to address this:

**Option 1:** Should `after-count` be assigned a default value (task result default) in case `count-comments-after` fails to execute?

**Option 2:** Allow specifying explicit default, e.g.:

```
     params:
        - name: after-count
          value: $(tasks.count-comments-after.results.count)
          default: "0"
```

**Option 3:** Do not execute `check-count` but continue executing rest of the final tasks.

Reply:
added a section `Known Limitations` with task results and pipeline results, [here](https://github.com/tektoncd/pipeline/pull/2437/files#diff-db9ac78ce08367ea2b7db8c1412049b6R658)